spec/util/bundler_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'spec_helper'
+
+RSpec.describe PDK::Util::Bundler do
+  before(:each) do
+    allow(PDK::CLI::Exec).to receive(:execute).and_return({exit_code: 0})
+  end
+
+  describe '.ensure_bundle!' do
+    context 'when there is no Gemfile' do
+      before(:each) do
+        allow(File).to receive(:file?).with(/Gemfile$/).and_return(false)
+      end
+
+      it 'does nothing' do
+        expect(PDK::CLI::Exec).to_not receive(:execute)
+
+        PDK::Util::Bundler.ensure_bundle!
+      end
+    end
+
+    context 'when there is no Gemfile.lock' do
+      before(:each) do
+        allow(File).to receive(:file?).with(/Gemfile$/).and_return(true)
+        allow(File).to receive(:file?).with(/Gemfile\.lock$/).and_return(false)
+      end
+
+      it 'generates Gemfile.lock' do
+        expect(PDK::CLI::Exec).to receive(:execute).with(/bundle(\.bat)?$/, 'lock', any_args).and_return({exit_code: 0})
```

Comment:
Yeah that would be simpler at this point, I think the original tests were written before the refactor that added the `PDK::CLI::Exec.bundle` method.